react/src/utils/Type.js

```diff
@@ -0,0 +1,66 @@
+// -------------------------------------------------------------------------------------------------
+// Miscellaneous type related functions.
+// -------------------------------------------------------------------------------------------------
+
+function IsString(value) {
+    return value !== undefined && value !== null && value.constructor === String;
+}
+
+function IsBoolean(value) {
+    return value !== undefined && value !== null && value.constructor === Boolean;
+}
+
+function IsNumber(value) {
+    return typeof(value) == 'number';
+}
+
+function IsObject(value) {
+    return value !== undefined && value !== null && value.constructor === Object;
+}
+
+function IsNonEmptyObject(value) {
+    return value !== undefined && value !== null && value.constructor === Object && Object.keys(value).length > 0;
+}
+
+function IsArray(value) {
+    return Array.isArray(value);
+}
+
+function IsNonEmptyArray(value) {
+    return Array.isArray(value) && value.length > 0;
+}
+
+function IsDateTime(value) {
+    return value instanceof Date;
+}
```

Comment:
I get the point in writing a lot of these but is this really any more readable than the actual body?

Reply:
Some of these aren't obvious (e.g. IsObject), so some I think are necessary, and for type checking consistency I thought similar functions for all type checking would be better.